projects/js-packages/ai-client/src/components/ai-control/block-ai-control.tsx

```diff
@@ -0,0 +1,278 @@
+/**
+ * External dependencies
+ */
+import { Button, ButtonGroup } from '@wordpress/components';
+import { useKeyboardShortcut } from '@wordpress/compose';
+import { useImperativeHandle, useRef, useEffect, useCallback, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import {
+	Icon,
+	closeSmall,
+	check,
+	arrowUp,
+	trash,
+	reusableBlock as regenerate,
+} from '@wordpress/icons';
+import debugFactory from 'debug';
+import React, { forwardRef } from 'react';
+/**
+ * Internal dependencies
+ */
+import { GuidelineMessage } from '../message/index.js';
+import AIControl from './ai-control.js';
+import './style.scss';
+/**
+ * Types
+ */
+import type { RequestingStateProp } from '../../types.js';
+import type { ReactElement } from 'react';
+
+type BlockAIControlProps = {
+	disabled?: boolean;
+	value: string;
+	placeholder?: string;
+	showAccept?: boolean;
+	acceptLabel?: string;
+	showButtonLabels?: boolean;
+	isTransparent?: boolean;
+	state?: RequestingStateProp;
+	showGuideLine?: boolean;
+	customFooter?: ReactElement;
+	onChange?: ( newValue: string ) => void;
+	onSend?: ( currentValue: string ) => void;
+	onStop?: () => void;
+	onAccept?: () => void;
+	onDiscard?: () => void;
+	showRemove?: boolean;
+	bannerComponent?: ReactElement;
+	errorComponent?: ReactElement;
+};
+
+const debug = debugFactory( 'jetpack-ai-client:ai-control' );
+
+/**
+ * BlockAIControl component. Used by the AI Assistant block, adding logic and components to the base AIControl component.
+ *
+ * @param {BlockAIControlProps} props  - Component props
+ * @param {React.MutableRefObject} ref - Ref to the component
+ * @returns {ReactElement}             Rendered component
+ */
+export function BlockAIControl(
```

Comment:
Are we planning to use this in more than one place? Could be a discussion about keeping it in the package or not (not for this PR)

Reply:
That is a very good catch. I'd also prefer to have this on the ai-assistant folder instead of the package.
Maybe we can have a call with @lhkowalski to decide where to move this and the other components as a last task?

Reply:
> Maybe we can have a call with @lhkowalski to decide where to move this and the other components as a last task?

Yes, for sure. Let's do it.